src/webapi/formdata.rs

```diff
@@ -0,0 +1,207 @@
+use crate::Value;
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webapi::blob::Blob;
+use webapi::element::Element;
+
+/// The `FormData` interface provides a way to easily construct a set of key/value pairs
+/// representing form fields and their values, which can then be easily sent using the
+/// `XMLHttpRequest.send()` method. It uses the same format a form would use if the encoding type
+/// were set to `"multipart/form-data"`.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/FormData)
+// https://xhr.spec.whatwg.org/#formdata
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "FormData")]
+pub struct FormData( Reference );
+
+impl FormData {
+    /// Creates a new `FormData`.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData)
+    pub fn new( form: &Element ) -> Self {
```

Comment:
From what I can see in the specs this doesn't take any `HTMLElement`, it requires `HTMLFormElement`, and the contructor can also be called without any parameters, and it can throw `InvalidStateError`.

But you don't have to add all of that if you don't need it; we can keep this function *if* we change its prototype to something like this:

```rust
pub fn from_element<T>( form: &T ) -> Result< Self, FormDataFromElementError > where T: IElement
```

where the `FormDataFromElementError` is an enum (use the `error_enum_boilerplate!` macro) with `InvalidStateError` and `TypeError` variants.

So, take the argument by its trait (since we currently don't have any `Deref`s; that might change in the future, but not yet.), and actually handle the cases where it either throws an `InvalidStateError`, or the passed `form` isn't actually an `HTMLFormElement`.